l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -212,6 +225,156 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
         $.operators[msg.sender].validatorCount -= uint128(validatorCount);
     }
 
+    /**
+     * @inheritdoc IUniFiAVSManager
+     * @dev Restricted in this context is like `whenNotPaused` modifier from Pausable.sol
+     */
+    function registerValidatorsOptimistically(ValidatorRegistrationParams[] memory paramsArray)
+        external
+        registeredOperator(msg.sender)
+        restricted
+    {
+        UniFiAVSStorage storage $ = _getUniFiAVSManagerStorage();
+        uint256 newValidatorCount = 0;
+        bytes memory delegateKey = $.operators[msg.sender].commitment.delegateKey;
+
+        for (uint256 i = 0; i < paramsArray.length; i++) {
+            ValidatorRegistrationParams memory params = paramsArray[i];
```

Comment:
This way it is more readable, but if in the function signature you use calldata, and use `paramsArray[i].whateverYouNeedFromStruct` you'll avoid copying calldata to memory, and the tx will be cheaper